src/commons/controlBar/ControlBarShareButton.tsx

```diff
@@ -66,12 +89,36 @@ export class ControlBarShareButton extends React.PureComponent<ControlBarShareBu
                   onChange={this.handleChange}
                   style={{ width: 175 }}
                 />
+                <>{console.log(this.props.programConfig)}</>
                 <ControlButton
                   label="Get Link"
                   icon={IconNames.SHARE}
                   onClick={() => {
-                    this.props.handleShortenURL(this.state.keyword);
-                    this.setState({ isLoading: true });
+                    // post request to backend, set keyword as return uuid
+                    const requestBody = {
+                      shared_program: {
+                        data: this.props.programConfig
+                      }
+                    };
+                    const fetchOpts: RequestInit = {
+                      method: 'POST',
+                      body: JSON.stringify(requestBody),
+                      headers: {
+                        'Content-Type': 'application/json'
+                      }
+                    };
+                    fetch('http://localhost:4000/api/shared_programs', fetchOpts)
+                      .then(res => {
+                        return res.json();
+                      })
+                      .then(resp => {
+                        this.setState({
+                          keyword: 'http://localhost:8000/playground/share/' + resp.uuid
+                        });
+                        console.log(resp);
+                      })
+                      .catch(err => console.log('Error: ', err));
```

Comment:
Ditto on debugging code. Also, we should show a warning if things go wrong instead of just logging it.

Reply:
Removed and changed to `.catch(err => showWarningMessage('fail to generate url! ' + err));`